packages/grid/src/vaadin-grid-selection-column.js

```diff
@@ -149,6 +162,81 @@ class GridSelectionColumn extends GridColumn {
     checkbox.indeterminate = this.__indeterminate;
   }
 
+  /** @private */
+  __lassoAutoScroller() {
+    if (this.__lassoDragStartIndex !== undefined) {
+      // Get the row being hovered over
+      const renderedRows = this._grid._getRenderedRows();
+      let rowHeight = 30;
+      const hoveredRow = renderedRows.find((row) => {
+        const rowRect = row.getBoundingClientRect();
+        rowHeight = rowRect.height;
+        return this.__lassoCurrentY >= rowRect.top && this.__lassoCurrentY <= rowRect.bottom;
+      });
+
+      // Get the index of the row being hovered over or the first/last
+      // visible row if hovering outside the grid
+      let hoveredIndex = hoveredRow ? hoveredRow.index : undefined;
+      const gridRect = this._grid.getBoundingClientRect();
+      if (this.__lassoCurrentY < gridRect.top) {
+        hoveredIndex = this._grid._firstVisibleIndex;
+      } else if (this.__lassoCurrentY > gridRect.bottom) {
+        hoveredIndex = this._grid._lastVisibleIndex;
+      }
+
+      if (hoveredIndex !== undefined) {
+        // Select all items between the start and the current row
+        renderedRows.forEach((row) => {
+          if (
+            (hoveredIndex > this.__lassoDragStartIndex &&
+              row.index >= this.__lassoDragStartIndex &&
+              row.index <= hoveredIndex) ||
+            (hoveredIndex < this.__lassoDragStartIndex &&
+              row.index <= this.__lassoDragStartIndex &&
+              row.index >= hoveredIndex)
+          ) {
+            if (this.__lassoSelect) {
+              this._grid.selectItem(row._item);
+            } else {
+              this._grid.deselectItem(row._item);
+            }
+          }
+        });
+      }
+
+      // Auto scroll the grid
+      this._grid.$.table.scrollTop += (this.__lassoDy || 0) / (rowHeight / 2);
+
+      // Schedule the next auto scroll
+      setTimeout(() => this.__lassoAutoScroller(), 100);
```

Comment:
The interval of `100ms` results in a pretty choppy scrolling. I would suggest to reduce this to at least 10-20ms.

Reply:
Done